csi-wrapper: add example for AWS EBS CSI driver (#1364)
confidential-containers/cloud-api-adaptor: volumes/csi-wrapper/examples/aws/README.md

# AWS EBS CSI Wrapper for Peer Pod Storage

## Prerequisites

> * Running Kubernetes cluster running on AWS Version >= 1.20

Reviewer: Maybe "Running Kubernetes cluster (version >= 1.20) on AWS" ?

* Peer-Pods is [deployed](../../../../aws/README.md)

## AWS EBS CSI Driver Installtion

> **NOTE:** the following is just a basic example, follow official [installtion instructions](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md) for advanced configuration.

Reviewer: s/installtion/installation/

```
aws iam create-policy \
	--policy-name EBS_Policy \
	--policy-document file://example-iam-policy.json
```
2.  Grant the driver IAM permissions:
```
kubectl create secret generic aws-secret \
    --namespace kube-system \
    --from-literal "key_id=${AWS_ACCESS_KEY_ID}" \
    --from-literal "access_key=${AWS_SECRET_ACCESS_KEY}"
```
3. Deploy the driver:
```
kubectl apply -k "github.com/kubernetes-sigs/aws-ebs-csi-driver/deploy/kubernetes/overlays/stable/?ref=release-1.21"
```
4. Verify the pods are running:
```
kubectl get pods -n kube-system -l app.kubernetes.io/name=aws-ebs-csi-driver
```

## Apply the PeerPods CSI wrapper

1. Create the PeerpodVolume CRD object
```
kubectl apply -f ../../crd/peerpodvolume.yaml
```
2. Apply RBAC roles to permit the wrapper to execute the required operations
```
kubectl apply -f rbac-ebs-csi-wrapper-runner.yaml
kubectl apply -f rbac-ebs-csi-wrapper-podvm.yaml
```
3. Patch the EBS CSI Driver:
```
kubectl patch deploy ebs-csi-controller -n kube-system --patch-file patch-controller.yaml
kubectl -n kube-system delete replicaset -l app=ebs-csi-controller
kubectl patch ds ebs-csi-node -n kube-system --patch-file patch-node.yaml
```

> 4. Verify the pods are running (each pod should contain an additional container):

Reviewer: I didn't understand `each pod should contain an additional container`. I see the pods running and I assumed it is fine.

Author: hmm,  for example if previously pod showed 3/3 containers, after patching it should show 4/4, suggestions for better wording are welcomed

> 4. Verify the pods are running (each pod should contain one more container than before the patching):

Reviewer 2: nit
```suggestion
4. Verify the pods are running (each pod should contain an additional container):
```
